p11-kit/list-objects.c

```diff
@@ -202,27 +270,59 @@ print_object (p11_list_printer *printer,
 	};
 	CK_ULONG n_attrs = sizeof (attrs) / sizeof (attrs[0]);
 
+	/* get the attributes */
+	p11_kit_iter_get_attributes (iter, arrs, n_arrs);
+	for (i = 0; i < n_arrs; ++i) {
+		if (arrs[i].ulValueLen == CK_UNAVAILABLE_INFORMATION)
+			continue;
+		arrs[i].pValue = malloc (arrs[i].ulValueLen + 1);
```

Comment:
I guess we can safely assume that ulValueLen accounts for NUL byte (otherwise we would need a overflow check here).

Reply:
I dont know about that. It feels safer to use the overflow check. I didnt find anywhere that a RFC2279 string would be NULL terminated.

Reply:
Yeah, you are right according to the PKCS#11 spec. Let's add a check.